src/util/tubemap.js

```diff
@@ -3109,26 +3116,129 @@ function drawNodes(dNodes) {
     .style("stroke-width", "2px")
     .append("svg:title")
-    .text((d) => getPopUpText(d));
+    .text((d) => getPopUpNodeText(d));
 }
 
-function getPopUpText(node) {
-
-  return (
-    `Node ID: ${node.name}` + (node.switched ? ` (reversed)` : ``) + `\n` +
-    `Node Length: ${node.sequenceLength} bases\n` +
-    `Haplotypes: ${node.degree}\n` +
-    `Aligned Reads: ${
-      node.incomingReads.length +
-      node.internalReads.length +
-      node.outgoingReads.length
-    }`
-  );
+function getPopUpNodeText(node) {
+  return `Node ID: ${node.name}` + (node.switched ? ` (reversed)` : ``) + `\n`;
+}
+
+// Get any node object by name.
+function getNodeByName(nodeName) {
+  if (typeof nodeName !== "string") {
+    throw new Error("Node Names must be strings");
+  }
+  // We just do a scan.
+  // TODO: index!
+  console.log("All nodes:", nodes); //
+  for (let i = 1; i < nodes.length; i++) {
+    // changes i to start from 1 instead of 0
+    if (nodes[i].name === nodeName) {
+      console.log("Found node with name", nodeName, "at index ", i);
+      return nodes[i];
+    }
+  }
+}
+
+function nodeSingleClick() {
+  /* jshint validthis: true */
+  // Get the node name
+  const nodeName = d3.select(this).attr("id");
+  let currentNode = getNodeByName(nodeName);
+  console.log("Node ", nodeName, " is ", currentNode);
+  if (currentNode === undefined) {
+    console.error("Missing node: ", nodeName);
+    return;
+  }
+  let nodeAttributes = [];
+  nodeAttributes.push([
+    "Node ID:",
+    currentNode.name + currentNode.switched ? "(reversed)" : "",
+  ]);
+  nodeAttributes.push(["Node Length:", currentNode.sequenceLength + " bases"]);
+  nodeAttributes.push(["Haplotypes:", currentNode.degree]);
+  nodeAttributes.push([
+    "Aligned Reads:",
+    currentNode.incomingReads.length +
+      currentNode.internalReads.length +
+      currentNode.outgoingReads.length,
+  ]);
+  nodeAttributes.push([
+    "Number of Reads Visiting Node:",
+    numReadsVisitNode(currentNode),
+  ]);
+  nodeAttributes.push(["Coverage:", coverage(currentNode, reads)]);
+
+  console.log("Single Click");
+  console.log("node show info callback", config.showInfoCallback);
+  config.showInfoCallback(nodeAttributes);
+}
+
+export function numReadsVisitNode(node) {
+  let countReads = new Set();
+  // incoming reads are reads that enter the node but don't start within it. They are represented as
+  // an array of subarrays which have 2 elements: an index indicating the read index and read path's index.
+  // The first node will not have any incoming reads.
+  for (let readVisit of node.incomingReads) {
+    countReads.add(readVisit[0]);
+  }
+  // internal reads are reads that start and end within the node. They are represented as
+  // an array of values which indicate the read index.
+  for (let read of node.internalReads) {
+    countReads.add(read);
+  }
+  // outgoing reads are reads that exit the node when the read starts within it. They are represented as
```

Comment:
Yeah this is probably the comment we need. I just didn't understand that it was actually true.